### PR TITLE
Adding a couple of actors to help migrate quagga to frr

### PR DESCRIPTION
**Pull request by mruprich**
Quagga has been replaced by FRR on RHEL 8 systems:
    https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/setting-your-routing-protocols_configuring-and-managing-networking

Actors automatically migrate the Quagga setup to FRR, except on systems using
the babeld daemon which has been dropped from FRR because of licensing issues.
If the use of the babeld daemon is detected, IPU is inhibited and user
is notified to reconfigure the system to use otherservices instead.

### CODE REVIEW

**Comment by mruprich** [comment]
leapp-ci build

File: repos/system_upgrade/el7toel8/actors/quaggareport/actor.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                        'babeld daemon which was a part of quagga implementation in RHEL7 '
+                        'is not available in RHEL8 in FRR due to licensing issues.'
+                    ),
+                    reporting.Severity(reporting.Severity.HIGH),
+                    reporting.Tags(COMMON_REPORT_TAGS),
+                    reporting.Flags([reporting.Flags.INHIBITOR]),
+                    reporting.Remediation(hint='Please use RIP, OSPF or EIGRP instead of Babel')
+                ])
+        else:
+            self.log.debug('babeld not used, upgrade may continue')
```
Comment (fellipeh) [inline repos/system_upgrade/el7toel8/actors/quaggareport/actor.py:40]: It's make no sense to me inform the user that upgrade may continue, that's was expected, or I misunderstood the point here. :)

File: repos/system_upgrade/el7toel8/models/quaggatofrrfacts.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from leapp.models import Model, fields
+from leapp.topics import SystemInfoTopic
+
+
+class QuaggaToFrrFacts(Model):
+    '''
+    A list of configuration files used by quagga. This list is used to add yes/no to
+    /etc/frr/daemons file. It indicates which daemons from frr should be run.
+    '''
```
Comment (fellipeh) [inline repos/system_upgrade/el7toel8/models/quaggatofrrfacts.py:9]: Please use right Doctring convention ( https://www.python.org/dev/peps/pep-0257/ )

File: repos/system_upgrade/el7toel8/actors/quaggareport/actor.py (hunk 1 above)
Comment (fellipeh) [reply repos/system_upgrade/el7toel8/actors/quaggareport/actor.py:40]: Thinking here, maybe just change the phrase "upgrade may continue"...

**Comment by bocekm** [comment]
@mruprich, do you plan to finish this PR? Can we help you with it?

**Comment by mruprich** [comment]
@bocekm Hi, sorry, I pushed this into backlog to finish other things. I am definitely planning to finish this.

**Comment by mruprich** [comment]
leapp-ci build

**Comment by ZhukovGreen** [comment]
@mruprich Hi Michal, we recently applied new naming convention to actors.
Actors library.py should be renamed to {actor_name}.py. Also test modules should contain the actor names inside, I.e. test_unit_myactor.py. Take a look to a few other actors.

**Comment by mruprich** [comment]
@fellipeh Hi, I am not sure how to cancel the request... Maybe you are the one who needs to do it? Anyway, I couldn't come up with anything smart. But these are just debug logs right?

**Comment by bocekm** [comment]
To get rid of the "change requested" from @fellipeh, the best would be if @fellipeh re-reviews and approves the code.

**Comment by bocekm** [comment]
@mruprich, is this still WIP or ready for a complete review?

**Comment by mruprich** [comment]
@bocekm I think it is ready now. It still needs some unit tests but we agreed with @pirat89 that I will write some tests after we agree that this is OK.

**Comment by mruprich** [comment]
I pushed some tests and some fixes that I noticed while writing the tests. These actors are quite simple so the tests are not very complex. But I think this should do it.

**Comment by mruprich** [comment]
@ZhukovGreen Hi, looking at the error - what should be the correct path for files in actor/tests/files? In the tests I tried relative paths but that always failed. So in the vagrantbox I tried absolute paths and that worked, but apparently not here. :(

**Comment by ZhukovGreen** [comment]
@mruprich Hi Michal, you solved the path problem well. Tomorrow, I will take a look at your MR

**Comment by ZhukovGreen** [comment]
I am sorry Michal @mruprich , today I stuck with some other task, so postponing the review to tomorrow. Thanks for understanding:)

File: repos/system_upgrade/el7toel8/actors/quaggadaemons/libraries/quaggadaemons.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    return True
+
+
+def process_daemons():
+    active_daemons = [daemon for daemon in QUAGGA_DAEMONS if _check_service(daemon, 'active')]
+    enabled_daemons = [daemon for daemon in QUAGGA_DAEMONS if _check_service(daemon, 'enabled')]
+
+    if active_daemons:
+        api.current_logger().debug('active quaggadaemons: {}'.format(', '.join(active_daemons)))
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggadaemons/libraries/quaggadaemons.py:32]: this is not the proper way of variable strings in logging. 

```diff
     if active_daemons:
-        api.current_logger().debug('active quaggadaemons: {}'.format(', '.join(active_daemons)))
+        api.current_logger().debug('active quaggadaemons: %s', ', '.join(active_daemons))
 
     if enabled_daemons:
-        api.current_logger().debug('enabled quaggadaemons: {}'.format(', '.join(enabled_daemons)))
+        api.current_logger().debug('enabled quaggadaemons: %s', ', '.join(enabled_daemons))
```

Check this for further information https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/#comment-4769289809

File: repos/system_upgrade/el7toel8/actors/quaggadaemons/tests/test_unit_quaggadaemons.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import errno
+import os
+
+from leapp.libraries.stdlib import api
+from leapp.libraries.actor import quaggadaemons
+from leapp.models import QuaggaToFrrFacts
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggadaemons/tests/test_unit_quaggadaemons.py:6]: pls check unused imports and remove them

File: repos/system_upgrade/el7toel8/actors/quaggatofrr/tests/test_unit_quaggatofrr.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import errno
+import os
+
+from leapp.libraries.stdlib import api
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggatofrr/tests/test_unit_quaggatofrr.py:4]: also unused imports

File: repos/system_upgrade/el7toel8/actors/quaggadaemons/actor.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+from leapp.actors import Actor
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+from leapp.models import InstalledRedHatSignedRPM, QuaggaToFrrFacts
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.actor.quaggadaemons import process_daemons
+
+
+class QuaggaDaemons(Actor):
+    """
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggadaemons/actor.py:9]: Please use multiline docstrings convention (ref https://www.python.org/dev/peps/pep-0257/#id17)

File: repos/system_upgrade/el7toel8/actors/quaggareport/actor.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+COMMON_REPORT_TAGS = [
+    reporting.Tags.NETWORK,
+    reporting.Tags.SERVICES
+]
+
+
+class QuaggaReport(Actor):
+    """
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggareport/actor.py:14]: Please use multiline docstrings convention (ref https://www.python.org/dev/peps/pep-0257/#id17)

File: repos/system_upgrade/el7toel8/actors/quaggatofrr/actor.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
+from leapp.actors import Actor
+from leapp.models import QuaggaToFrrFacts
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
+from leapp.libraries.actor.quaggatofrr import process_facts
+
+
+class QuaggaToFrr(Actor):
+    """
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggatofrr/actor.py:8]: Please use multiline docstrings convention (ref https://www.python.org/dev/peps/pep-0257/#id17)

File: repos/system_upgrade/el7toel8/models/quaggatofrrfacts.py [hunk 9]
```diff
@@ -0,0 +1,13 @@
+from leapp.models import Model, fields
+from leapp.topics import SystemInfoTopic
+
+
+class QuaggaToFrrFacts(Model):
+    """
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/models/quaggatofrrfacts.py:6]: Please use multiline docstrings convention (ref https://www.python.org/dev/peps/pep-0257/#id17)

File: repos/system_upgrade/el7toel8/actors/quaggadaemons/libraries/quaggadaemons.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    'ripd',
+    'ripngd',
+    'zebra'
+]
+
+
+def _check_service(name, state):
+    try:
+        run(['systemctl', 'is-{}'.format(state), name])
+        api.current_logger().debug('{} is {}'.format(name, state))
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggadaemons/libraries/quaggadaemons.py:19]: this is not the proper way of variable strings in logging. See the comments bellow

File: repos/system_upgrade/el7toel8/actors/quaggadaemons/libraries/quaggadaemons.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    except CalledProcessError:
+        api.current_logger().debug('{} is not {}'.format(name, state))
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggadaemons/libraries/quaggadaemons.py:21]: this is not the proper way of variable strings in logging. See the comments bellow

File: repos/system_upgrade/el7toel8/actors/quaggadaemons/libraries/quaggadaemons.py [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+def process_daemons():
+    active_daemons = [daemon for daemon in QUAGGA_DAEMONS if _check_service(daemon, 'active')]
+    enabled_daemons = [daemon for daemon in QUAGGA_DAEMONS if _check_service(daemon, 'enabled')]
+
+    if active_daemons:
+        api.current_logger().debug('active quaggadaemons: {}'.format(', '.join(active_daemons)))
+
+    if enabled_daemons:
+        api.current_logger().debug('enabled quaggadaemons: {}'.format(', '.join(enabled_daemons)))
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggadaemons/libraries/quaggadaemons.py:35]: this is not the proper way of variable strings in logging. See the comments above

File: repos/system_upgrade/el7toel8/actors/quaggatofrr/libraries/quaggatofrr.py [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+# due to an error in quagga, the conf files are not deleted after uninstall
+# we can copy them as they are
+def _copy_config_files(src_path, dest_path):
+    conf_files = os.listdir(src_path)
+    for file_name in conf_files:
+        full_path = os.path.join(src_path, file_name)
+        if os.path.isfile(full_path):
+            shutil.copy(full_path, dest_path)
+            api.current_logger().debug('Copying {} to {}{}'.format(full_path, dest_path, file_name))
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggatofrr/libraries/quaggatofrr.py:81]: this is not the proper way of variable strings in logging. See the comments above

File: repos/system_upgrade/el7toel8/actors/quaggadaemons/actor.py [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+    Checking for daemons that are currently running in the system.
+    The tools will check for config files later on since these should stay in the system.
+    """
+
+    name = 'quagga_daemons'
+    consumes = (InstalledRedHatSignedRPM,)
+    produces = (QuaggaToFrrFacts, )
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggadaemons/actor.py:16]: please be consistent. Better remove space

File: repos/system_upgrade/el7toel8/actors/quaggadaemons/actor.py [hunk 15]
```diff
@@ -0,0 +1,21 @@
+from leapp.actors import Actor
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+from leapp.models import InstalledRedHatSignedRPM, QuaggaToFrrFacts
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.actor.quaggadaemons import process_daemons
+
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggadaemons/actor.py:6]: wrong order of imports. Please check also other modules for that. FYI there is a tool isort for this purpose

**Comment by mruprich** [comment]
Thanks @ZhukovGreen, the 7e92d0b fixes requested changes. Regarding the test for quaggareport - I honestly did not think that this piece would need a test. It is only a report. It does not even have a library or anything. What do you think?

**Comment by ZhukovGreen** [comment]
@mruprich tests are never a waste of time:) see https://github.com/oamg/leapp-repository/pull/467/commits/7a0b8fe28e3ce4e42245ca804cb6d6146d433348

**Comment by ZhukovGreen** [comment]
@mruprich Currently, the coverage has improved, but still some places need to be covered and are important to cover. Let me know if you need a help with it 
```
----------- coverage: platform linux, python 3.8.3-final-0 -----------
Name                                                                            Stmts   Miss  Cover   Missing
-------------------------------------------------------------------------------------------------------------
repos/system_upgrade/el7toel8/actors/quaggadaemons/actor.py                        13      2    85%   23-24
repos/system_upgrade/el7toel8/actors/quaggadaemons/libraries/quaggadaemons.py      19      7    63%   17-24
repos/system_upgrade/el7toel8/actors/quaggareport/actor.py                         20      1    95%   45
repos/system_upgrade/el7toel8/actors/quaggatofrr/actor.py                          13      3    77%   20-23
repos/system_upgrade/el7toel8/actors/quaggatofrr/libraries/quaggatofrr.py          52     22    58%   48-56, 64-70, 76-81, 85-87
-------------------------------------------------------------------------------------------------------------
TOTAL                                                                             171     35    80%
```

**Comment by ZhukovGreen** [comment]
@mruprich give me 15 more min to work on your MR, I'll ping you. I found some things to fix from my previous commit

**Comment by mruprich** [comment]
Thanks @ZhukovGreen I'll try to expand the tests.

**Comment by ZhukovGreen** [comment]
In the meantime I will try to fix the issue with the capturing the log:) just can't bypass it:))))

**Comment by ZhukovGreen** [comment]
Should work after https://github.com/oamg/leapp/pull/652 will be merged

**Comment by ZhukovGreen** [comment]
hi @mruprich , I fixed the issue with the capturing logs. The solution was... do not capture it:) because it is not working and randomly frezzes the test session. So I fixed it in a more reliable way. Let me know if you need help with the increasing coverage.

**Comment by mruprich** [comment]
@ZhukovGreen I re-did the tests for quagga-to-frr actor and added one more test. With your test(thank you by the way) for quagga-report, the test base should be pretty decent I think.

**Comment by ZhukovGreen** [comment]
leapp-ci build

**Comment by ZhukovGreen** [comment]
Could anyone from @oamg/qe check why is https://url.corp.redhat.com/a5f47fe failing? I tried, howver I don't see anything except the derpecation warning, which shouldn't cause the failure

**Comment by ZhukovGreen** [comment]
@mruprich I added couple commits, to cover some last islands of not covered code. it is shining now:)

**Review by ZhukovGreen — DISMISSED** [review]

**Comment by bocekm** [comment]
I can see in the last e2e test result that this happened in the initramfs. Artem thinks it's already fixed. He'll rerun the tests:
```
[  322.049581] upgrade[532]: 2020-08-03 18:28:53.299 INFO     PID: 3 leapp.workflow.Applications: Executing actor quagga_to_frr
[  322.092407] upgrade[532]: Process Process-164:
[  322.094251] upgrade[532]: Traceback (most recent call last):
[  322.096022] upgrade[532]:   File "/usr/lib64/python3.6/multiprocessing/process.py", line 258, in _bootstrap
[  322.098815] upgrade[532]:     self.run()
[  322.099956] upgrade[532]:   File "/usr/lib64/python3.6/multiprocessing/process.py", line 93, in run
[  322.102536] upgrade[532]:     self._target(*self._args, **self._kwargs)
[  322.104466] upgrade[532]:   File "/root/tmp_leapp_py3/leapp/repository/actor_definition.py", line 72, in _do_run
[  322.107392] upgrade[532]:     actor_instance.run(*args, **kwargs)
[  322.109115] upgrade[532]:   File "/root/tmp_leapp_py3/leapp/actors/__init__.py", line 335, in run
[  322.111612] upgrade[532]:     self.process(*args)
[  322.112962] upgrade[532]:   File "/etc/leapp/repos.d/system_upgrade/el7toel8/actors/quaggatofrr/actor.py", line 20, in process
[  322.116824] upgrade[532]:     quagga_facts = next(self.consume(QuaggaToFrrFacts))
[  322.118948] upgrade[532]: StopIteration
```

**Comment by bocekm** [comment]
This PR will need to be verified. I'll start a discussion about who should be the QE engineer to verify it/create an upgrade test.

**Comment by ZhukovGreen** [comment]
e2e tests

**Review by fellipeh — DISMISSED** [review]
LGTM

- upgrade was successfully made
- tests passed

File: repos/system_upgrade/el7toel8/actors/quaggareport/actor.py [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        try:
+            quagga_facts = next(self.consume(QuaggaToFrrFacts))
+        except StopIteration:
+            return
+        if 'babeld' in quagga_facts.active_daemons or 'babeld' in quagga_facts.enabled_daemons:
+            create_report([
+                reporting.Title('Babeld is not available in FRR'),
```
Comment (shaded-enmity) [inline repos/system_upgrade/el7toel8/actors/quaggareport/actor.py:33]: Can we provide a little bit detail about FRR here via link `reporting.ExternalLink(url='https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/setting-your-routing-protocols_configuring-and-managing-networking', title='Setting routing protocols in RHEL8')`

**Comment by pirat89** [comment]
any progress around the verification? who is working on it?

**Comment by psklenar** [comment]
> @oamg/qe any progress around the verification? who is working on it?
https://issues.redhat.com/browse/OAMG-4001?focusedCommentId=15310026&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15310026

**Comment by mruprich** [comment]
Just rebasing to latest master.

**Comment by mruprich** [comment]
IMO the error is not related to this actor.

**Comment by AloisMahdal** [comment]
...aaand, we got traceback in the quagga_to_frr in the initramfs:

      [..]
      [..]
    [  456.105984] upgrade[623]: 2020-10-20 06:37:01.74  INFO     PID: 3 leapp.workflow.Applications: Starting stage Main of phase Applications
    [  456.108350] upgrade[623]: 2020-10-20 06:37:01.77  INFO     PID: 3 leapp.workflow.Applications: Executing actor migrate_brltty
    [  456.151788] upgrade[623]: 2020-10-20 06:37:01.130 INFO     PID: 3 leapp.workflow.Applications: Executing actor network_manager_update_config
    [  456.202180] upgrade[623]: 2020-10-20 06:37:01.178 INFO     PID: 411 leapp.workflow.Applications.network_manager_update_config: Consuming dhcp=
    [  456.207920] upgrade[623]: 2020-10-20 06:37:01.185 INFO     PID: 411 leapp.workflow.Applications.network_manager_update_config: Written the following to /usr/lib/NetworkManager/conf.d/10-dhcp-dhclient.conf:
    [  456.211275] upgrade[623]: # Generated by leapp when upgrading from RHEL7 to RHEL8
    [  456.212952] upgrade[623]: [main]
    [  456.214025] upgrade[623]: dhcp=dhclient
    [  456.229507] upgrade[623]: 2020-10-20 06:37:01.207 INFO     PID: 3 leapp.workflow.Applications: Executing actor quagga_to_frr
    [  456.281199] upgrade[623]: Process Process-162:
    [  456.282596] upgrade[623]: Traceback (most recent call last):
    [  456.284036] upgrade[623]:   File "/usr/lib64/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    [  456.286050] upgrade[623]:     self.run()
    [  456.287198] upgrade[623]:   File "/usr/lib64/python3.6/multiprocessing/process.py", line 93, in run
    [  456.289094] upgrade[623]:     self._target(*self._args, **self._kwargs)
    [  456.290639] upgrade[623]:   File "/root/tmp_leapp_py3/leapp/repository/actor_definition.py", line 72, in _do_run
    [  456.292697] upgrade[623]:     actor_instance.run(*args, **kwargs)
    [  456.294168] upgrade[623]:   File "/root/tmp_leapp_py3/leapp/actors/__init__.py", line 335, in run
    [  456.296034] upgrade[623]:     self.process(*args)
    [  456.297298] upgrade[623]:   File "/etc/leapp/repos.d/system_upgrade/el7toel8/actors/quaggatofrr/actor.py", line 20, in process
    [  456.299531] upgrade[623]:     quagga_facts = next(self.consume(QuaggaToFrrFacts))
    [  456.301199] upgrade[623]: StopIteration
    [  456.304965] upgrade[623]: ==============================================================================================
    [  456.307161] upgrade[623]: Actor quagga_to_frr unexpectedly terminated with exit code: 1 - Please check the above details
    [  456.309310] upgrade[623]: ==============================================================================================
    [  456.311447] upgrade[623]: 2020-10-20 06:37:01.282 INFO     PID: 3 leapp: Answerfile will be created at /var/log/leapp/answerfile
    [  456.355800] upgrade[623]: Debug output written to /var/log/leapp/leapp-upgrade.log
    [  456.357631] upgrade[623]: ============================================================
    [  456.359374] upgrade[623]:                            REPORT
    [  456.360782] upgrade[623]: ============================================================
    [  456.362515] upgrade[623]: A report has been generated at /var/log/leapp/leapp-report.json
    [  456.364264] upgrade[623]: A report has been generated at /var/log/leapp/leapp-report.txt
    [  456.366016] upgrade[623]: ============================================================
    [  456.367744] upgrade[623]:                        END OF REPORT
    [  456.369173] upgrade[623]: ============================================================
    [  456.370882] upgrade[623]: Answerfile has been generated at /var/log/leapp/answerfile
    [  456.418050] XFS (vda1): Unmounting Filesystem
      [..]
      [..]

**Comment by mruprich** [comment]
@ZhukovGreen Hi, can you please review this instead of @pirat89 ? Not much has changed really, just one short commit but all in all it is the same as before.

File: repos/system_upgrade/el7toel8/actors/quaggatofrr/tests/test_unit_quaggatofrr.py [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+        os.mkdir(TO_DIR)
+
+        for num in range(1, 10):
+            full_path = "{}test_file_{}".format(FROM_DIR, num)
+            with open(full_path, 'w') as fp:
+                fp.write("test_file_{}".format(num))
+        yield
+    finally:
+        shutil.rmtree(FROM_DIR)
+        shutil.rmtree(TO_DIR)
```
Comment (ZhukovGreen) [inline repos/system_upgrade/el7toel8/actors/quaggatofrr/tests/test_unit_quaggatofrr.py:35]: No action required, just FYI:

@mruprich it is safer implementation now. If someone in future will use this function, it is a chance he will forget to cleanup.  So it is usually better to implement teardown via context managers or fixtures with the context managers (https://docs.pytest.org/en/stable/fixture.html#fixture-finalization-executing-teardown-code)

It is also worse to mention that pytest provides you a fixture to create temp dirs (https://docs.pytest.org/en/stable/tmpdir.html), but it is just FYI as well

**Review by ZhukovGreen — APPROVED** [review]
@mruprich Please address the @shaded-enmity concern, the rest is fine to me. Thx

**Comment by mruprich** [comment]
@ZhukovGreen @AloisMahdal This should do it....

**Review by ZhukovGreen — APPROVED** [review]
Excellent

**Comment by AloisMahdal** [comment]
Verified with:

    leapp-0.11.1-100.202010201731Z.287d684.master.el7_6
    leapp-repository-0.12.0-0.202010221359Z.7b66953.quagga_to_frr_migration.PR467

 From QE side, this is good to merge.

**Comment by ZhukovGreen** [comment]
@pirat89 Petr, any wishes before the merge? I assume to squash it and merge

File: repos/system_upgrade/el7toel8/actors/quaggareport/actor.py (hunk 16 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/quaggareport/actor.py:33]: it has been addressed

**Comment by pirat89** [comment]
@mruprich @ZhukovGreen no, just the squash with the good commit message. I went through the code to create one. Hopefully it's correct. I am going to merge this with the squash. Updating the description of the PR as well.

File: repos/system_upgrade/el7toel8/actors/quaggareport/actor.py (hunk 1 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/quaggareport/actor.py]: has been resolved. closing discussion

File: repos/system_upgrade/el7toel8/models/quaggatofrrfacts.py (hunk 2 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/models/quaggatofrrfacts.py]: has been resolved. closing discussion

**Comment by pirat89** [comment]
(just guys, next time please close conversations when resolved :-) I do not have go through that then to see whether something has been forgotten or whether it's resolved)

**Pull request merged**